chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -6,8 +6,14 @@ import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.atomic.AtomicBoolean
 
 internal class RoomStore(
-        private val roomsMap: MutableMap<String, Room> = ConcurrentHashMap()
+        private val roomsMap: MutableMap<String, Room> = ConcurrentHashMap(),
+        private val roomsPassedInitialState: MutableList<String> = arrayListOf()
```

Comment:
I think we should keep this state and the place we consume it together if possible, so we could either move this property to the RoomService, or move the check in to here (by returning an additional boolean from `applyMembershipEvent` which tells us whether to `emit` the events or not).

Reply:
Additionally, I think a `MutibleSet` would be a better match for recording which rooms have received an initial state in the past. The ordering of List is not useful to us, and neither is the fact that a list can hold duplicate items.

Reply:
Good suggestions! I've refactored the `roomsPassedInitialState` to the `RoomService` and made it a mutable set :)